address.go

```diff
@@ -364,7 +368,7 @@ func (a *AddressScriptHash) ScriptAddress() []byte {
 // IsForNet returns whether or not the pay-to-script-hash address is associated
 // with the passed bitcoin network.
 func (a *AddressScriptHash) IsForNet(net *chaincfg.Params) bool {
-	return bytes.Equal(a.netID, net.ScriptHashAddrID)
+	return bytes.Equal(a.netID, net.ScriptHashAddrID) && a.cksumHasher == net.Base58CksumHasher
```

Comment:
Why is IsForNet only for Base58CksumHasher?

Reply:
I'm not sure that I understood you question correctly. The reason why I've added this check is because 2 networks can have the same `netID`, but different hashers. Eg. Bitcoin testnet address `mvbu1Gdy8SUjTenqerxUaZyYjmveZvt33q` and Groestlcoin testnet address `mvbu1Gdy8SUjTenqerxUaZyYjmvedc787y` have the same `netID`. Since `IsForNet` takes  `*chaincfg.Params` parameter (rather than `netID` only) I assumed it should return false if called for Groestlcoin Testnet address with Bitcoin Testnet `*chaincfg.Params` because Groestlcoin Testnet address is not valid Bitcoin Testnet address. I hope this answers your question.

Reply:
I understand your point. 
However, I think the function should return true for any valid Groestlcoin address, even if there is a possible collision with some other coin. We always use one Blockbook instance per coin, so for given instance we are testing Groestlcoins, not Bitcoins. 
Even though it is not 100% correct, I think returning false would confuse users of the function more.

Reply:
Do you mean you want to compare `netID` only as in original code and ignore hasher? OK, I can revert this part if you want so.
But, I think it will be more confusing for user: if user works with Groestlcoin, but passes Bitcoin parameters, it means he is doing something wrong and we should let him know by returning false, rather than silently ignoring mistake and return true.

Reply:
I've pushed commit to revert this check.

Reply:
Yes, I agree that the best would be to check netID and also the checksum. 
But I think that for our purpose it is fine as it is.